New Let's Encrypt intermediate CAs

brianhlin opened the pull request:
https://letsencrypt.org/2020/11/06/own-two-feet.html

brianhlin committed:
Remove unused .txt suffixes

brianhlin committed:
Add new Let's Encrypt intermediate CAs

https://letsencrypt.org/2020/11/06/own-two-feet.html

brianhlin commented:
@jbasney it looks like Let's Encrypt provides the `.pem` directly so this also includes changes to the Makefile but I'm not sure if you had other reasons for including the `.txt` suffixes. FWIW, we're removing the suffix in our own packaging: https://github.com/opensciencegrid/osg-certificates/blob/master/build-certificates-dir.sh#L19

jbasney commented:
Thanks @brianhlin - I will review and merge.

brianhlin committed:
Update Let's Encrypt Root CA signing policy for new intermediates

brianhlin committed:
Forgot to escape the single-quotes

brianhlin committed:
Forgot to update sources/targets in the Makefile

brianhlin commented:
Thanks @jbasney! @ddavila0 gave me an offline review, hence the fixup commits.

jbasney commented on lets-encrypt-r3.pem:
```diff
@@ -0,0 +1,30 @@
+-----BEGIN CERTIFICATE-----
```
This is a derived file installed by the Makefile using `$(GET) https://letsencrypt.org/certs/lets-encrypt-r3.pem` so I don't think we want it committed to the repo.

brianhlin replied:
Removed

jbasney commented on lets-encrypt-r4.pem:
```diff
@@ -0,0 +1,30 @@
+-----BEGIN CERTIFICATE-----
```
This is a derived file installed by the Makefile using `$(GET) https://letsencrypt.org/certs/lets-encrypt-r4.pem` so I don't think we want it committed to the repo.

brianhlin replied:
Removed

jbasney commented on Makefile:
```diff
@@ -23,41 +28,61 @@ install : all
 
 clean :
-	$(RM) *.0 *.signing_policy *.pem.txt
+	$(RM) *.0 *.signing_policy *.pem
```
This change is fine, but note that this `make clean` rule relies on using `.signing_policy.txt` suffixes for the sources so it only deletes the `*.signing_policy` derived files, not the `*.signing_policy.txt` sources.

brianhlin replied:
Ah, I missed this detail in my rush! This should be good to go now

jbasney commented on Makefile:
```diff
@@ -1,13 +1,18 @@
-sources = isrgrootx1.signing_policy.txt \
-          letsencryptauthorityx3.signing_policy.txt \
-          letsencryptauthorityx4.signing_policy.txt
+sources = isrgrootx1.signing_policy \
```
Dropping the `.txt` suffix means these sources will be deleted by the `make clean` rule below.

brianhlin replied:
Fixed with the updated `make clean` in https://github.com/cilogon/letsencrypt-certificates/pull/3/commits/c40bc6cbd76460bfc2d5f3894cac731e457b1051

jbasney reviewed (changes requested):
See comments inline.

edquist commented on Makefile:
```diff
@@ -1,13 +1,18 @@
-          letsencryptauthorityx3.signing_policy.txt \
-          letsencryptauthorityx4.signing_policy.txt
+sources = isrgrootx1.signing_policy \
+          letsencryptauthorityx3.signing_policy \
+          letsencryptauthorityx4.signing_policy \
+	  lets-encrypt-r3.signing_policy \
+	  lets-encrypt-r4.signing_policy \
```
Whitespace

```suggestion
          letsencryptauthorityx3.signing_policy \
          letsencryptauthorityx4.signing_policy \
          lets-encrypt-r3.signing_policy \
          lets-encrypt-r4.signing_policy \
```

brianhlin replied:
I saw the weirdness in the diff yesterday and was a little confused because the lines visually lined up in my editor/on GitHub. It looks like my editor's adding tabs instead of spaces in Makefiles but spaces everywhere else!

edquist commented on Makefile:
```diff
@@ -1,13 +1,18 @@
-          23c2f850.0 4042bcee.0 4a0a35c0.0 4f06f81d.0 6187b673.0 929e297e.0 \
-          isrgrootx1.pem.txt \
-          letsencryptauthorityx3.pem.txt letsencryptauthorityx4.pem.txt
+	  8d33f237.signing_policy dec71a0b.signing_policy \
+	  9f194ecd.signing_policy dd7d39a7.signing_policy \
+          23c2f850.0 4042bcee.0 4a0a35c0.0 4f06f81d.0 6187b673.0 8d33f237.0 929e297e.0 9f194ecd.0 dec71a0b.0 dd7d39a7.0 \
+          isrgrootx1.pem \
+          letsencryptauthorityx3.pem letsencryptauthorityx4.pem \
+	  lets-encrypt-r3.pem lets-encrypt-r4.pem
```
Whitespace, and respect upstream's attempt to preserve 80-wide here

```suggestion
          8d33f237.signing_policy dec71a0b.signing_policy \
          9f194ecd.signing_policy dd7d39a7.signing_policy \
          23c2f850.0 4042bcee.0 4a0a35c0.0 4f06f81d.0 6187b673.0 \
          8d33f237.0 929e297e.0 9f194ecd.0 dec71a0b.0 dd7d39a7.0 \
          isrgrootx1.pem \
          letsencryptauthorityx3.pem letsencryptauthorityx4.pem \
          lets-encrypt-r3.pem lets-encrypt-r4.pem
```

brianhlin replied:
I've split this up slightly differently to group the relevant hashes together

edquist reviewed (commented):
@brianhlin - small details noticed for when you do your next set of changes

brianhlin committed:
Replace tabs with spaces

brianhlin committed:
Split derived X.509 certs into multiple lines

brianhlin committed:
Remove LE intermediates that are pulled at install time

brianhlin committed:
Ensure that we only clean up derived files from 'make install'

brianhlin committed:
More whitespace fixups

brianhlin commented:
Hi @jbasney, I believe I've addressed all your comments. Let me know if you want me to clean this PR up with a rebase or if you prefer to leave the details of the sausage-making in place :)
